sakura_core/charset/charcode.h

```diff
@@ -27,60 +27,47 @@
 #pragma once
 
 //2007.09.13 kobake 作成
+#include <array>
 #include "parse/CWordParse.h"
 #include "util/std_macro.h"
 
-// -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
-//                         判定関数                            //
-// -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
-// #include "charset/codechecker.h"
-// SJIS関連コードは codecheker.hに移動
-
 // -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
 //                           定数                              //
 // -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- //
 // SJISのコードページ(CP_ACP では無くこれを使えばおそらく英語版Winでも動くはず。)	2008/5/12 Uchi
-#define CP_SJIS		932
-
-//定数の素 (直接使用は控えてください)
-#define TAB_ 				'\t'
-#define SPACE_				' '
-#define CR_					'\015'
-#define LF_					'\012'
-#define ESC_				'\x1b'
-#define CRLF_				"\015\012"
+constexpr int CP_SJIS = 932;
 
 //UNICODE定数
 namespace WCODE{
 	//文字
-	static const wchar_t TAB   = LCHAR(TAB_);
-	static const wchar_t SPACE = LCHAR(SPACE_);
-	static const wchar_t CR    = LCHAR(CR_);
-	static const wchar_t LF    = LCHAR(LF_);
-	static const wchar_t ESC   = LCHAR(ESC_);
+	constexpr wchar_t TAB   = L'\t';
+	constexpr wchar_t SPACE = L' ';
+	constexpr wchar_t CR    = L'\015';
+	constexpr wchar_t LF    = L'\012';
+	constexpr wchar_t ESC   = L'\x1b';
 
 	//文字列
-	static const wchar_t CRLF[] = LTEXT(CRLF_);
+	constexpr wchar_t CRLF[] = L"\015\012";
```

Comment:
```suggestion
	constexpr const wchar_t CRLF[] = L"\r\n";
```
GCCではconstexpr付けても`wchar_t[]`らしくて警告が出るんだそうです。
あと、8進数の文字コード読める人はレアなので普通にエスケープシーケンスにしたらいいと思います。

Reply:
- `constexpr const` に改めました。
- エスケープシーケンスにしました。